### DISCUSSION TITLE
Screen based editor (was INSERT enhancement)

### DISCUSSION THREAD

**Discussion by leventp** [paulscottrobson/neo6502-firmware#189] [Implemented Ideas]
At the moment, INSERT (CTRL+E) enables insertion of one character at a time. When editing a BASIC program line, this is a bit cumbersome. If possible, it would be desirable to keep INSERT mode on until another control code is entered.

**Comment by paulscottrobson** [comment]
Slightly problematic because of the way the insert works, but not impossible. Though I'd rather accept this as is, and work on the screen editor design which would behave normally.

**Comment by leventp** [comment]
Great. Looking forward to this.

**Comment by paulscottrobson** [comment]
I think it will be a generic editor for everything, so it would have an API to request line text, insert, delete and so on. Not quite sure yet.

**Comment by leventp** [comment]
One thing I like about the current screen/line editor is that it is very flexible (Commodore 64 like) and I can go anywhere on the screen make changes and apply them instantly. Can issue a cat command and load any file on the screen (with multiple inserts to make room for the load command). I suggest the new editor should be invoked by a command (say 'edit') so we won't lose the existing editing capabilities.

**Reply by erlendnagel** [reply]
I agree that a standalone editor would be better. For one thing, it would probably allow for larger files to be edited.

**Comment by paulscottrobson** [comment]
The plan is to have an API driven editor with individual horizontal scrolling lines. I think Action does this, it's slightly odd. The original idea from this comes from the 6800 Sphere firmware, which has an editor design which is somewhat novel.  It will allow editing by just providing and updating lines on demand, thus it can be used for anything, and any enhancements would work the same way. It's still in the planning stage though :)
